tests/layers/CMakeLists.txt

```diff
@@ -114,3 +68,33 @@ add_vk_layer(wrap_objects ${WRAP_SRCS})
 
 set(TEST_SRCS test.cpp vk_layer_table.cpp vk_layer_extension_utils.cpp)
 add_vk_layer(test ${TEST_SRCS})
+
+if(WIN32 OR CMAKE_GENERATOR MATCHES "^Xcode.*")
```

Comment:
You're really trying to determine if you have a multi-config generator here, and this test mixes up platform and generator.  Also, I'm not sure if there isn't a way to do a WIN32/Windows build on something other than Visual Studio and if that way uses a single-config or multi-config generator.

It may be better to check to see if the CMAKE_CONFIGURATION_TYPES list is empty or not, where not empty means multi-config and that this path must be taken.

Note that there is a new GENERATOR_IS_MULTI_CONFIG property that is more reliable, but it does not show up until CMake 3.9.

I do recognize that CMake code elsewhere in this repo and other repos that use the approach used here or something like it, but perhaps we could start migrating to something better in new code.

And if checking CMAKE_CONFIGURATION_TYPES is too uncomfortable, then maybe checking for MSVC instead of WIN32 might be better.

Reply:
I wrote this as a way of simplifying the old logic, but you're right that checking for Windows or XCode isn't the cleanest solution. The GENERATOR_IS_MULTI_CONFIG property sounds perfect here, except for the version restrictions. But I do think your solution of checking if its Visual Studio or XCode at least sounds better than the logic I currently have. I'll toy around with it and see what seems to work best.